shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1676,59 +1676,60 @@ - (CGRect)firstRectForRange:(UITextRange*)range {
         startSelectionRect = _selectionRects[i].rect;
-      } else {
-        return _selectionRects[i].rect;
       }
-    }
-    if (!CGRectIsNull(startSelectionRect)) {
-      minY = fmin(minY, CGRectGetMinY(_selectionRects[i].rect));
-      maxY = fmax(maxY, CGRectGetMaxY(_selectionRects[i].rect));
-      BOOL endsOnOrAfterEndOfRange = _selectionRects[i].position >= end - 1;  // end is exclusive
-      BOOL nextSelectionRectIsOnNextLine =
-          !isLastSelectionRect &&
-          // Selection rects from different langauges in 2 lines may overlap with each other.
-          // A good approximation is to check if the center of next rect is below the bottom of
-          // current rect.
-          // TODO(hellohuanlin): Consider passing the line break info from framework.
-          CGRectGetMidY(_selectionRects[i + 1].rect) > CGRectGetMaxY(_selectionRects[i].rect);
-      if (endsOnOrAfterEndOfRange || isLastSelectionRect || nextSelectionRectIsOnNextLine) {
-        endSelectionRect = _selectionRects[i].rect;
-        break;
+      if (!CGRectIsNull(startSelectionRect)) {
+        minY = fmin(minY, CGRectGetMinY(_selectionRects[i].rect));
+        maxY = fmax(maxY, CGRectGetMaxY(_selectionRects[i].rect));
+        BOOL endsOnOrAfterEndOfRange = _selectionRects[i].position >= end - 1;  // end is exclusive
+        BOOL nextSelectionRectIsOnNextLine =
+            !isLastSelectionRect &&
+            // Selection rects from different langauges in 2 lines may overlap with each other.
+            // A good approximation is to check if the center of next rect is below the bottom of
+            // current rect.
+            // TODO(hellohuanlin): Consider passing the line break info from framework.
+            CGRectGetMidY(_selectionRects[i + 1].rect) > CGRectGetMaxY(_selectionRects[i].rect);
+        if (endsOnOrAfterEndOfRange || isLastSelectionRect || nextSelectionRectIsOnNextLine) {
+          endSelectionRect = _selectionRects[i].rect;
+          break;
+        }
       }
     }
-  }
-  if (CGRectIsNull(startSelectionRect) || CGRectIsNull(endSelectionRect)) {
-    return CGRectZero;
+    if (CGRectIsNull(startSelectionRect) || CGRectIsNull(endSelectionRect)) {
+      return CGRectZero;
+    } else {
+      // fmin/fmax to support both LTR and RTL languages.
+      CGFloat minX = fmin(CGRectGetMinX(startSelectionRect), CGRectGetMinX(endSelectionRect));
+      CGFloat maxX = fmax(CGRectGetMaxX(startSelectionRect), CGRectGetMaxX(endSelectionRect));
+      return CGRectMake(minX, minY, maxX - minX, maxY - minY);
+    }
   } else {
-    // fmin/fmax to support both LTR and RTL languages.
-    CGFloat minX = fmin(CGRectGetMinX(startSelectionRect), CGRectGetMinX(endSelectionRect));
-    CGFloat maxX = fmax(CGRectGetMaxX(startSelectionRect), CGRectGetMaxX(endSelectionRect));
-    return CGRectMake(minX, minY, maxX - minX, maxY - minY);
+    // Auto correction highlight on iOS 16 and older is rendered by Flutter.
+    return CGRectZero;
```

Comment:
The initial implementation of firstRectForRange that returns the first rectangle (which by the way was incorrect) was introduced for the scribble feature on iPad (code [here](https://github.com/flutter/engine/commit/1ae721c0230e8e3f3618f7e9d301ad2133b2fd61#diff-4c7b102c0690b8ec5e2212b079f5d69fe3f816c84e47ce94bc7bc89312f39e40R1487-R1505)). It turns out that the change was not necessary for scribble - I have verified that keep returning CGRectZero doesn't break scribble.